lib/plugins/TaskRunner/BitbucketDownloader.js

```diff
@@ -0,0 +1,125 @@
+   */
+  constructor(container, options) {
+    super(container, options)
+
+    var build = this.build = options.build
+    var project = this.project = options.project
+
+    if(project.provider.type != "bitbucket"){
+      throw new Error("Unsupported provider type: " + project.provider.type)
+    }
+
+    this.params = {
+      provider_slug: project.provider.slug,
+      owner: project.owner,
+      repo: project.repo,
+      token: project.service_auth.token,
+      refreshToken: project.service_auth.refreshToken,
+      ref: options.ref || build.ref,
+      format: 'tar.gz',
+    }
+
+    // delay run function until we've done an async action
+
+    var run = this.run
+    var self = this
+    /** 
+     * Overrides AbstractPlugin::run to lookup OAuth1.0 credentials before running actual task.
+     */
+    this.run = Promise.promisify(function(cb){
+      self.log.debug(`making auth request to bitbucket handler for ${self.description()}`)
+
+      // response
+      //  - auth: To be used as the value of the Authorization header
+      //  - url: Full request URL (provider host + path in the lookup request)
+      self.getNewToken(function(err, response){
+        if(err){
+          self.log.error({err, project: self.project}, `Failed to get Bitbucket credentials: ${err.message}`)
+
+          // write the error to the process stdout
+          self.setScript([
+            `echo "Failed to get Bitbucket credentials: ${err.message}"`
+          ])
+        }
+        else {
+          self.log.debug(`auth request to bitbucket handler success for ${self.description()}`)
+
+          self.params.auth = response.auth
+          self.params.url = response.url
+
+          // build the shell script now that we have all the auth/request info
+          self.setBitbucketScript()
+
+          // filter out secret strings (the full auth header)
+          self.options.secrets = [
+            response.auth,
+          ]
+        }
+
+        return run(cb)
+      })
+    })
+  }
+
+  setBitbucketScript(){
+    var token = this.token;
+    var params = this.params;
+    var script = [
+      'mkdir -p $SRC_DIR',
+      'cd $SRC_DIR',
+      `wget -q -O - --header 'Authorization:${token}' "${params.url}" | tar xzf - `
+    ]
+
+    this.setScript(script)  // Script::setScript()
+  }
+
+  description(){
+    return `${this.plugin} ${this.params.owner}/${this.params.repo} @ ${this.params.ref}`
+  }
+
+  getNewToken(done) {
+    var self = this;
+
+    request.post(this.prepareRefresh(), function(error, response, body) {
+      // TODO: HANDLE ERRORS
+      self.token = body.access_token;
+      done();
+    });
+  };
+
+  prepareRefresh(){
+    return {
+      url: "https://bitbucket.org/site/oauth2/access_token",
+      form: {
+        grant_type: 'refresh_token',
+        refresh_token: this.params.refreshToken,
+      },
+      auth: {
+        user: config.bbh.hook_key,
```

Comment:
I know it's a bit confusing because `probo/defaults.yaml` has GHH config information, but that's for the GHH handler, not the Container Manager. These two will be split soon.